torch/distributed/distributed_c10d.py

```diff
@@ -1700,7 +1701,7 @@ def gather_object(obj, object_gather_list=None, dst=0, group=None):
         object_gather_list[i] = _tensor_to_object(tensor, tensor_size)
 
 
-def broadcast_object_list(object_list, src=0, group=None):
+def broadcast_object_list(object_list, src=0, group=None, map_location=None):
```

Comment:
Would the rest of the APIs benefit from `map_location`? If so feel free to file issues for those (maybe we can bootcamp them)